CHANGELOG.md

```diff
@@ -28,6 +28,7 @@ scheme are considered to be bugs.
 
 ### Fixed
 
+* [#500](https://github.com/hashie/hashie/pull/500): Do not warn when setting Mash keys that look like underbang, bang, and query methods. However, still warn when setting a Mash key that looks like a setter due to method access for that key being a syntax error and colliding with the setter method for a similar key - [@michaelherold](https://github.com/michaelherold).
```

Comment:
For changelog I would remove all starting from "However". If it's long, it needs to be in README or UPGRADING.